### PR TITLE
New ES Alignment: alcareco output modules, ESAlignment put into stand…

### PR DESCRIPTION
**Pull request by shervin86**
…ard sequences

@emanueledimarco and @alphatsai and @cmkuo would like to follow that

New ES Alignment ALCARECO:
- event size: 60 kB/ev

### CODE REVIEW

**Comment by mmusich** [comment]
please test

**Comment by diguida** [comment]
@shervin86 with this PR you have just introduced the AlCaReco for ES alignment, and associated it to a PD, but you have not added it to the release validation matrix. Are you planning to do it into a different PR?

**Comment by shervin86** [comment]
It should be done in this PR, can you point me to the workflows and files to be modified?

I suppose I have to modify 
Configuration/PyReleaseValidation/python/relval_standard.py

-- 
 							Shervin

**Comment by mmusich** [comment]
@shervin86 please rebase it. It now conflicts with my PR for the updated AlCaMatrix #12754.
For adding the AlCaReco into the Validation matrix you can modify [relval_steps.py](https://github.com/cms-sw/cmssw/blob/CMSSW_8_0_X/Configuration/PyReleaseValidation/python/relval_steps.py)

**Comment by mmusich** [comment]
-1 
doesn't merge

**Comment by mmusich** [comment]
@shervin86 is there any plan to rebase this one soon?

**Comment by shervin86** [comment]
Doing it now

-- 
 							Shervin

**Comment by mmusich** [comment]
please test

File: Calibration/EcalAlCaRecoProducers/python/ALCARECOEcalESAlign_Output_cff.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        # 'keep recoTracks_ecalAlCaESAlignTrackReducer_*_*',
+        # 'keep recoTrackExtras_ecalAlCaESAlignTrackReducer_*_*',
+        # 'keep TrackingRecHitsOwned_ecalAlCaESAlignTrackReducer_*_*',
+        # 'keep SiStripClusteredmNewDetSetVector_ecalAlCaESAlignTrackReducer_*_*',
+        # 'keep SiPixelClusteredmNewDetSetVector_ecalAlCaESAlignTrackReducer_*_*',
+
+        # 'keep ESDigiCollection_ecalPreshowerDigis_*_*',
+        # 'keep ESDCCHeaderBlocksSorted_ecalPreshowerDigis_*_*',
+        # 'keep ESKCHIPBlocksSorted_ecalPreshowerDigis_*_*',
+
```
Comment (mmusich) [inline Calibration/EcalAlCaRecoProducers/python/ALCARECOEcalESAlign_Output_cff.py]: ```$ edmEventSize -v -a EcalESAlign.root ``

``File EcalESAlign.root Events 14
Branch Name | Average Uncompressed Size (Bytes/Event) | Average Compressed Size (Bytes/Event) 
ESDigiCollection_ecalPreshowerDigis__RECO. 14078.4 8043.86
recoTrackExtras_ecalAlCaESAlignTrackReducer__RECO. 5486.36 2955.36
TrackingRecHitsOwned_ecalAlCaESAlignTrackReducer__RECO. 17974.6 2436.71
SiStripClusteredmNewDetSetVector_ecalAlCaESAlignTrackReducer__RECO. 5047.29 2432.5
recoTracks_ecalAlCaESAlignTrackReducer__RECO. 4895.93 2066.14
ESKCHIPBlocksSorted_ecalPreshowerDigis__RECO. 21759.6 1403.36
SiPixelClusteredmNewDetSetVector_ecalAlCaESAlignTrackReducer__RECO. 2089.43 1184.86
ESDCCHeaderBlocksSorted_ecalPreshowerDigis__RECO. 12854.7 1020.21
EventProductProvenance 1677.14 211
recoBeamSpot_offlineBeamSpot__RECO. 496.071 182.143
EventAuxiliary 176.214 52.9286
EventSelections 133.571 17.8571
BranchListIndexes 62.2857 11.4286```

any reason not to keep the commented list with explicit module name?

Comment (shervin86) [inline Calibration/EcalAlCaRecoProducers/python/ALCARECOEcalESAlign_Output_cff.py]: 2 lines is better then 8 :-)
I would prefer to have the alcareco transparent to the changes of  ecalPreshowerDigis and ecalAlCaESAlignTrackReducer. So if in the future the collections produced by those two modules change, the alCaReco should not be modified accordingly.


-- 
 							Shervin

**Comment by shervin86** [comment]
Hi, is there anything preventing this PR to be signed?

**Comment by mmusich** [comment]
@shervin86 see my comments above. I think that's the preferred option to have explicit module names in the keep statements. @davidlange6 please confirm if this is the case.

**Comment by shervin86** [comment]
Ok, will do it.

-- 
 							Shervin

**Comment by mmusich** [comment]
@shervin86 any hope of getting this rebased and cleaned up for the next pre-release?

**Comment by mmusich** [comment]
@shervin86 since I haven't seen activity here I have created a new one that merges. #13203

**Pull request closed without merging**